client/src/lib/offline-storage.ts

```
@@ -1,163 +1,352 @@
+  async getAllClients(): Promise<OfflineClient[]> {
+    return db.clients.toArray();
+  }
+
+  async updateClient(id: number, updates: Partial<Omit<OfflineClient, 'id' | 'document' | 'synced' | 'serverId'>>): Promise<void> {
+    await db.clients.update(id, { ...updates, synced: false, timestamp: Date.now() });
+    const clientData = await db.clients.get(id);
+    if(clientData){ // Ensure clientData is not null
+      const syncData = { ...clientData, ...updates, localId: id }; // Use the full updated data for sync
+      // Check if an update_client task already exists, update it, otherwise add new.
+      const existingSyncItem = await db.syncQueue().where({localId: id, type: 'update_client'}).first();
+      if (existingSyncItem && existingSyncItem.id) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)